fitbenchmarking/fitbenchmarking
# Issue #347: Restructure directories

#### Description of Work

Fixes #347 

Makes the following changes:
fitbenchmarking/fitting/controllers -> fitbenchmarking/controllers
fitbenchmarking/fitting/tests -> fitbenchmarking/controllers/tests
fitbenchmarking/resproc -> fitbenchmarking/plotting
fitbenchmarking/fitting_benchmarking.py -> fitbenchmarking/core/fitting_benchmarking.py
fitbenchmarking/fitbenchmark_one_problem.py -> fitbenchmarking/core/fitbenchmark_one_problem.py
fitbenchmarking/results_output.py -> fitbenchmarking/core/results_output.py

#### Testing Instructions

1. Check these names are more intuitive.

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
